specs/WebMessageObjects.md

```diff
@@ -0,0 +1,176 @@
+# Background
+This is a proposal for a new API that will provide a framework for native representation of DOM
+objects from page content in WebView2 and its implementation for DOM File objects. The (specific ask
+from WebView2)[https://github.com/MicrosoftEdge/WebView2Feedback/issues/501] is to be able get to
+the paths for DOM file objects, which is not accessible to page content. We also have asks to
+expose other DOM objects, including iframes, <object> objects, etc. which will be added under this
+WebMessageObjects framework in future.
+
+# Conceptual pages (How To)
+
+WebMessageObjects are representations of DOM objects that can be passed via the [WebView2 WebMessage
+API](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2webmessagereceivedeventargs)
+to the app. You can examine supported DOM objects using native reflections of the types.
+
+Currently the only supported DOM object type with this API is:
+- [File](https://developer.mozilla.org/en-US/docs/Web/API/File)
+
+Page content in WebView2 can pass objects to the app via the
```

Comment:
The implied other half of this feature would be CoreWebView2.PostWebMessageAsJSONWithAdditionalObjects that would let you post files into the web content and so giving web content access to files it wouldn't otherwise be allowed. Its not required for the drag and drop feature, but makes sense for this general purpose feature you're describing here. Even if we're not going to implement it immediately for drag and drop, can we add it to this API review spec?